mldsa/poly.h

```diff
@@ -84,18 +84,31 @@ __contract__(
 );
 
 #define poly_use_hint MLD_NAMESPACE(poly_use_hint)
-void poly_use_hint(poly *b, const poly *a, const poly *h);
+void poly_use_hint(poly *b, const poly *a, const poly *h)
+__contract__(
+  requires(memory_no_alias(a,  sizeof(poly)))
+  requires(memory_no_alias(b, sizeof(poly)))
+  requires(memory_no_alias(h, sizeof(poly)))
+  requires(array_bound(a->coeffs, 0, MLDSA_N, 0, MLDSA_Q))
```

Comment:
If `array_bound(arr, idx_low, idx_high, value_low, value_high)` asserts that the values of the array `arr` within`[idx_low, ..., idx_high - 1]` are within the range `[value_low, ..., value_high]`, then for parameter `a`, shouldn't we use `requires(array_bound(a->coeffs, 0, MLDSA_N, 0, MLDSA_Q-1))` (also for `h->coeffs in [0;2)`)? 
I notice that mlkem-native uses a similar pattern with bounds `[0, MLKEM_Q]`. I'm wondering if there's something I'm missing about how these bounds should be specified.

Reply:
Thank you! I actually just checked the macro itself in the` cbmc.h` file and now I understand the proof contract. 
My confusion is based on the https://github.com/pq-code-package/mlkem-native/blob/main/proofs/cbmc/proof_guide.md which defines the `array_bound` as inclusive for both lower and upper bounds.

Reply:
Arrgh! Sorry.. that's a bug in proof_guide.md from earlier (when the upper-bound _was_ inclusive.). We changed it to exclusive upper bounds some time ago. I will raise a PR on the Proof Guide immediately.